php/elFinderVolumeGoogleDrive.class.php

```
@@ -0,0 +1,1659 @@
+		||  !$this->options['accessToken']) {
+			return $this->setError('Required options undefined.');
+		}		
+		
+		// make net mount key
+		$this->netMountKey = md5(join('-', array('googledrive', $this->options['path'])));
+		
+		if (! $this->oauth) {				
+				
+				$client = new Google_Client();
+								 				
+				$client->setAccessToken($this->session->get('elFinderGoogleDriveAuthTokens'));										
+				$this->oauth = new Google_Service_Drive($client);				
+		}
+		
+		if (! $this->oauth) {
+			return $this->setError('OAuth extension not loaded.');
+		}
+				
+		// normalize root path
+		if($this->options['path'] == '/' || $this->options['path'] == 'root'){
+			$this->options['path'] = '/';				
+		}	
+		$this->root = $this->options['path'] = $this->_normpath($this->options['path']);		
+		
+		$this->options['root'] == '' ?  $this->options['root']= 'GoogleDrive.com' : $this->options['root'];
+		
+		if (empty($this->options['alias'])) {
+			$this->options['alias'] = ($this->options['path'] === '/') || ($this->options['path'] === 'root')? $this->options['root'] : $this->getPathtoName($this->options['path']).'@GDrive';
+			
+		}
+
+		$this->rootName = $this->options['alias'];
+				
+		$this->googledriveUid = $this->options['googledriveUid'];
+		$this->tmbPrefix = 'googledrive'.base_convert($this->googledriveUid, 10, 32);
+		
+		if (!empty($this->options['tmpPath'])) {
+			if ((is_dir($this->options['tmpPath']) || @mkdir($this->options['tmpPath'])) && is_writable($this->options['tmpPath'])) {
+				$this->tmp = $this->options['tmpPath'];
+			}
+		}
+		if (!$this->tmp && is_writable($this->options['tmbPath'])) {
+			$this->tmp = $this->options['tmbPath'];
+		}
+		if (!$this->tmp && ($tmp = elFinder::getStaticVar('commonTempPath'))) {
+			$this->tmp = $tmp;
+		}
+		
+		if (!empty($this->options['metaCachePath'])) {
+			if ((is_dir($this->options['metaCachePath']) || @mkdir($this->options['metaCachePath'])) && is_writable($this->options['metaCachePath'])) {
```

Do not use the `@` to hide the error.